pkg/controller/v1beta1/inferenceservice/reconcilers/keda/keda_reconciler.go

```diff
@@ -0,0 +1,212 @@
+	corev1 "k8s.io/api/core/v1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("KedaReconciler")
+
+type KedaReconciler struct {
+	client       client.Client
+	scheme       *runtime.Scheme
+	ScaledObject *kedav1alpha1.ScaledObject
+	componentExt *v1beta1.ComponentExtensionSpec
+}
+
+func NewKedaReconciler(client client.Client,
+	clientset kubernetes.Interface,
+	scheme *runtime.Scheme,
+	componentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec) *KedaReconciler {
+	return &KedaReconciler{
+		client:       client,
+		scheme:       scheme,
+		ScaledObject: createKedaScaledObject(clientset, componentMeta, componentExt),
+		componentExt: componentExt,
+	}
+}
+
+func getKedaMetrics(metricConfig *v1beta1.MetricsConfig, metadata metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec) []kedav1alpha1.ScaleTriggers {
+	var triggers []kedav1alpha1.ScaleTriggers
+	var serverAddress string
+
+	// Default values
+	triggerType := string(corev1.ResourceCPU)
+	metricType := autoscalingv2.UtilizationMetricType
+	scaleTarget := int(constants.DefaultCPUUtilization)
+
+	// Get metric configuration from metricConfig (configmap)
+	if metricConfig != nil && metricConfig.MetricBackend != "" {
+		triggerType = metricConfig.MetricBackend
+		serverAddress = metricConfig.ServerAddress
+	}
+	// override metric configuration from componentExtension if it is set
+	if componentExt.ScaleMetric != nil {
+		triggerType = string(*componentExt.ScaleMetric)
+	}
+	if componentExt.ScaleMetricType != nil {
+		metricType = *componentExt.ScaleMetricType
+	}
+	if componentExt.ScaleTarget != nil {
+		scaleTarget = *componentExt.ScaleTarget
+	}
+	// override metric configuration from componentExtension.ScalerSpec if it is set
+	if componentExt.ScalerSpec != nil {
+		if componentExt.ScalerSpec.ScaleMetric != nil {
+			triggerType = string(*componentExt.ScalerSpec.ScaleMetric)
+		}
+		if componentExt.ScalerSpec.ServerAddress != "" {
+			serverAddress = componentExt.ScalerSpec.ServerAddress
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			scaleTarget = *componentExt.ScalerSpec.ScaleTarget
+		}
+		if componentExt.ScalerSpec.ScaleMetricType != nil {
+			metricType = *componentExt.ScalerSpec.ScaleMetricType
+		}
+	}
+
+	trigger := kedav1alpha1.ScaleTriggers{
+		Type:     triggerType,
+		Metadata: map[string]string{},
+	}
+
+	// set trigger metadata for prometheus and graphite triggers
+	if triggerType == "prometheus" || triggerType == "graphite" {
+		if serverAddress != "" {
+			trigger.Metadata["serverAddress"] = serverAddress
+		}
+		if componentExt.ScalerSpec != nil {
```

Comment:
Downstream (OCP) the namespace attribute is [required to be set](https://docs.openshift.com/container-platform/4.17/nodes/cma/nodes-cma-autoscaling-custom-trigger.html). Could you add that?